wallet/src/main/java/com/android/identity_credential/wallet/ui/TimeAgo.kt

```diff
@@ -0,0 +1,116 @@
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Format time interval to the time instant in the past in human-readable way, such as
+ * "3 days and 5 hours ago".
+ *
+ * [instant] time instant in the past
+ * [now] time now, only exposed for testing
+ */
+@Composable
+fun TimeAgo(instant: Instant, now: Instant = Clock.System.now()): String {
+    val tick = remember { mutableIntStateOf(0) }
+    val tickValue = tick.intValue
+    val duration = now.minus(instant)
+    val tz = TimeZone.currentSystemDefault()
+    val months = instant.monthsUntil(now, tz)
+    val updateAt: Instant
+    val text: String
+    if (months == 0) {
+        // No need for calendar calculations
+        val days = duration.inWholeDays
+        val hours = duration.inWholeHours
+        if (days != 0L) {
+            updateAt = instant.plus((hours + 1).hours)
+            val daysStr = pluralStringResource(R.plurals.time_ago_day, days.toInt(), days.toInt())
+            val extraHours = (hours - 24 * days).toInt()
+            text = if (extraHours == 0) {
+                daysStr
+            } else {
+                val hoursStr = pluralStringResource(R.plurals.time_ago_hour, extraHours, extraHours)
+                stringResource(R.string.time_ago_days_and_hours, daysStr, hoursStr)
+            }
+        } else if (hours != 0L) {
+            val minutes = duration.inWholeMinutes
+            updateAt = instant.plus((minutes + 1).minutes)
+            val extraMinutes = (minutes - 60 * hours).toInt()
+            val hoursStr = pluralStringResource(R.plurals.time_ago_hour, hours.toInt(), hours.toInt())
+            text = if (extraMinutes == 0) {
+                hoursStr
+            } else {
+                val minutesStr = pluralStringResource(R.plurals.time_ago_minute, extraMinutes, extraMinutes)
+                stringResource(R.string.time_ago_hours_and_minutes, hoursStr, minutesStr)
+            }
+        } else {
+            val minutes = duration.inWholeMinutes
+            val seconds = duration.inWholeSeconds
+            updateAt = instant.plus((seconds + 1).seconds)
+            text = if (minutes != 0L) {
+                val minutesStr = pluralStringResource(R.plurals.time_ago_minute, minutes.toInt(), minutes.toInt())
+                val extraSeconds = (seconds - 60 * minutes).toInt()
+                if (extraSeconds == 0) {
+                    minutesStr
+                } else {
+                    val secondsStr = pluralStringResource(R.plurals.time_ago_second, extraSeconds, extraSeconds)
+                    stringResource(R.string.time_ago_minutes_and_seconds, minutesStr, secondsStr)
+                }
+            } else if (seconds != 0L) {
+                pluralStringResource(R.plurals.time_ago_second, seconds.toInt(), seconds.toInt())
+            } else {
+                stringResource(R.string.time_ago_now)
+            }
+        }
```

Comment:
as a personal preference leaning towards modularity (and for legibility as well), I would have moved this code to a function, ex: `calculateTimeThisMonth()` and the logic in the `else` below in a function, ex:`calculcateTimeBeyondThisMonth()`

Reply:
There are two things that were at the back of my mind: one is that I'd need these functions also to be Composable -  but for no (IMHO) good reason, as it would only be needed to access resources. The other is that we also compute `updateAt`, not just `text`, and of course there is a way to return multiple values, but it also makes it a bit heavier. Perhaps there is a way to factor all these branches in a smarter way.

Reply:
I agree we can refactor this later _if_ there's a need to expose the underlying functions to the rest of the app. I doubt this will be the case.

FWIW, there are costs to premature modularization (similar to the costs of its cousin, Premature Optimization), for example, suppose we did expose `fun calcTimeAgoString(Duration duration): String` there's a risk someone would start using it in the UI and their UI wouldn't auto-update and we didn't catch this at code-review time. By forcing people to use the `TimeAgo` composable this bug can't happen.

Reply:
I modularized it, but along different axis.